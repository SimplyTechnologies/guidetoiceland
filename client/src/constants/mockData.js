let _id = 0;

const genId = () => ++_id;

function generateMock() {
	return [{
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}]
}

function shuffle(b) {
	let a = [...b];
    for (let i = a.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [a[i], a[j]] = [a[j], a[i]];
    }
    return a;
}


export const mockData = {
	'Popular tips': generateMock(),
	'Verified locals': shuffle(generateMock()),
	'Latest tips': shuffle(generateMock()),
	'Newest locals': shuffle(generateMock())
};