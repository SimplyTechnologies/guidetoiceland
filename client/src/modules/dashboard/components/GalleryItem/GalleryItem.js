import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'reactstrap';

function GalleryItem({ item }) {
	return (
		<Col
			xs={12}
			sm={6}
			md={4}
			lg={2}
			style={{
				background: `url(${item.src}) center center no-repeat`,
				height: '300px',
				backgroundSize: 'cover',
			}}	
			className={'col animated fadeInUp gallery-col'}
		>
			<div className={'gallery-item-title'}>
				<span>{item.title}</span>
			</div>
			<div className={'gallery-item-description'}>
				<span>{item.description}</span>
			</div>
		</Col>
	);
}

GalleryItem.propTypes = {
	item: PropTypes.shape({
		id: PropTypes.number.isRequired,
		src: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		description: PropTypes.string.isRequired
	})
};

export default GalleryItem;