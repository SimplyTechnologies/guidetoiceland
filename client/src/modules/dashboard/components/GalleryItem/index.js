import GallerItem from './GalleryItem';

export default GallerItem;