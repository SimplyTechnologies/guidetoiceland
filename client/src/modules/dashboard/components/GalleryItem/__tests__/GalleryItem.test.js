import React from 'react';
import renderer from 'react-test-renderer';
import GalleryItem from '../GalleryItem';
import { mockData } from 'constants/mockData';

const filter = Object.keys(mockData)[0];
const data = mockData[filter];
describe('GalleryItem', () => {
	let renderedInstance;
	beforeEach(() => {
		renderedInstance = renderer.create(
			<GalleryItem item={data[1]} />
		);
	});
	it('should render correctly', () => {
		expect(renderedInstance).toMatchSnapshot();
	});
});