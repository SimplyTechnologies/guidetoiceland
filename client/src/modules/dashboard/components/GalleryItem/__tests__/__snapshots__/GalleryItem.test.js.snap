// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem should render correctly 1`] = `
<div
  className="col animated fadeInUp gallery-col col-12 col-sm-6 col-md-4 col-lg-2"
  style={
    Object {
      "background": "url(https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed) center center no-repeat",
      "backgroundSize": "cover",
      "height": "300px",
    }
  }
>
  <div
    className="gallery-item-title"
  >
    <span>
      Lorem ipsum ahmist ahad
    </span>
  </div>
  <div
    className="gallery-item-description"
  >
    <span>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer.
    </span>
  </div>
</div>
`;
